### PR TITLE
Py-pymol package has been updated to the latest version

### PR DESCRIPTION
**Pull request by hseara**
*   I have removed the previous py-pymol version as it was no longer compiling
*   I have added a new package libmmtf-cpp that is a new dependency of py-pymol

### CODE REVIEW

**Comment by hseara** [comment]
Any problem with this submission, or I just have to wait for a reviewer?

File: var/spack/repos/builtin/packages/py-pymol/package.py [hunk 1]
```diff
@@ -13,20 +13,22 @@ class PyPymol(PythonPackage):
...
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('tcsh', type=('build', 'run'))
+    depends_on('py-pyqt5', type=('build', 'run'))
+    depends_on('py-pmw', type=('build', 'run'))
+    depends_on('mesa-glu', type=('build', 'run'))
+
+    depends_on('libmmtf-cpp', type=('build'))
+    depends_on('msgpack-c', type=('build'))
+
+    patch('apbstools_tcltk8.6.patch', level=0, when='@2.3.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pymol/package.py]: Can you explain this patch in more detail? Why is it necessary? Has it been submitted upstream?

Comment (hseara) [reply var/spack/repos/builtin/packages/py-pymol/package.py]: @adamjstewart I just copied the patch from the archlinux pymol installation. But a more careful inspection makes me believe that it is not needed. First, the plugin it tries to fix is outdated.  Also, we do not provide apbs support at this moment, so it is useless.

**Comment by hseara** [comment]
*  I have removed the patch which I believe is useless
*  I removed an uneeded dependency which is already by freeglut

**Pull request merged**